Dockerfile

```
@@ -107,10 +110,18 @@ RUN for PYBIN in /opt/python/*/bin; do \
 # **************************
 FROM install-pynini-from-wheel AS run-tests
 
+RUN echo "How does one run absl-py tests? We may never know the answer... Oh well."
+
 # Copy Pynini's tests and testing assets (but not Pynini itself)
-COPY --from=wheel-building-env /src/pynini-${PYNINI_VERSION}/pynini_test.py /tests/pynini_test.py
-COPY --from=wheel-building-env /src/pynini-${PYNINI_VERSION}/testdata /tests/testdata
+COPY --from=wheel-building-env /src/pynini-${PYNINI_VERSION}/tests /tests
+
+# RUN curl https://copr.fedorainfracloud.org/coprs/vbatts/bazel/repo/epel-7/vbatts-bazel-epel-7.repo > /etc/yum.repos.d/vbatts-bazel-epel-7.repo \
+#     && yum install -y bazel3
 
 # Run Pynini's tests for each of our Pythons
-RUN cd /tests && for PYBIN in /opt/python/*/bin; do \
-    "${PYBIN}/python" pynini_test.py -v -f || exit 1; done
+RUN cd / && for PYBIN in /opt/python/*/bin; do \
+    "${PYBIN}/pip" install absl-py || exit 1; \
+    for TEST in "tests/*_test.py"; do \
+        "${PYBIN}/python" ${TEST} ; \
```

```suggestion
        "${PYBIN}/python" ${TEST} || exit 1 ; \
```
You need this if you want ANY test failure to translate into build failures. As it is right now, only the last test's exit code will be seen by docker.

Minimal example: compare `for FOO in ls bs ls; do $FOO ; done;` (bad) with `for FOO in ls bs ls; do $FOO || exit 1 ; done;` (good) on your local machine.